transmit/direct_transmit_test.go

```diff
@@ -0,0 +1,502 @@
+		assert.Equal(t, "error when sending event", errorEvent.Fields["error"])
+		assert.Equal(t, http.StatusInternalServerError, errorEvent.Fields["status_code"])
+		assert.Contains(t, errorEvent.Fields, "roundtrip_usec")
+	})
+}
+
+func TestDirectTransmission(t *testing.T) {
+	testServer := newTestDirectAPIServer(t)
+
+	mockMetrics := &metrics.MockMetrics{}
+	mockMetrics.Start()
+
+	// Use max batch size of 3 for testing
+	dt := NewDirectTransmission(mockMetrics, "test", 3, 50*time.Millisecond)
+	dt.Logger = &logger.NullLogger{}
+	dt.Version = "test-version"
+	dt.Transport = http.DefaultTransport.(*http.Transport)
+
+	err := dt.Start()
+	require.NoError(t, err)
+	defer dt.Stop()
+
+	now := time.Now().UTC()
+
+	// Create events for multiple datasets and scenarios
+	var allEvents []*types.Event
+
+	// Dataset A: 5 events (should create 2 batches: 3 + 2)
+	for i := range 5 {
+		eventData := types.NewPayload(map[string]any{
+			"trace.trace_id": fmt.Sprintf("trace-a-%d", i),
+			"dataset":        "A",
+			"event_id":       i,
+		})
+
+		event := &types.Event{
+			Context:     context.Background(),
+			APIHost:     testServer.server.URL,
+			APIKey:      "api-key-a",
+			Dataset:     "dataset-a",
+			Environment: "test",
+			SampleRate:  10,
+			Timestamp:   now.Add(time.Duration(i) * time.Millisecond),
+			Data:        eventData,
+		}
+		allEvents = append(allEvents, event)
+	}
+
+	// Dataset B: 4 events (should create 2 batches: 3 + 1)
+	for i := range 4 {
+		eventData := types.NewPayload(map[string]any{
+			"trace.trace_id": fmt.Sprintf("trace-b-%d", i),
+			"dataset":        "B",
+			"event_id":       i,
+		})
+
+		event := &types.Event{
+			Context:     context.Background(),
+			APIHost:     testServer.server.URL,
+			APIKey:      "api-key-b",
+			Dataset:     "dataset-b",
+			Environment: "test",
+			SampleRate:  20,
+			Timestamp:   now.Add(time.Duration(i+100) * time.Millisecond),
+			Data:        eventData,
+		}
+		allEvents = append(allEvents, event)
+	}
+
+	// Dataset C: 2 events (should create 1 batch via timeout)
+	for i := range 2 {
+		eventData := types.NewPayload(map[string]any{
+			"trace.trace_id": fmt.Sprintf("trace-c-%d", i),
+			"dataset":        "C",
+			"event_id":       i,
+		})
+
+		event := &types.Event{
+			Context:     context.Background(),
+			APIHost:     testServer.server.URL,
+			APIKey:      "api-key-c",
+			Dataset:     "dataset-c",
+			Environment: "test",
+			SampleRate:  30,
+			Timestamp:   now.Add(time.Duration(i+200) * time.Millisecond),
+			Data:        eventData,
+		}
+		allEvents = append(allEvents, event)
+	}
+
+	// Send all events
+	for _, event := range allEvents {
+		dt.EnqueueEvent(event)
+	}
+
+	// Wait for all events to be processed (11 total)
+	require.Eventually(t, func() bool {
+		receivedEvents := testServer.getEvents()
+		return len(receivedEvents) == 11
+	}, 200*time.Millisecond, 10*time.Millisecond)
+
+	// Verify all events were received
+	receivedEvents := testServer.getEvents()
```

Comment:
What's the difference between this check and the require.Eventually check above?

Reply:
that is just some AI slop!